conformance-test.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+venv() {
+  if ! test -d "$1" ; then
+	  if command -v virtualenv > /dev/null; then
+      virtualenv -p python3 "$1"
+	  else
+	    python3 -m venv "$1"
+	  fi
+  fi
+  source "$1"/bin/activate
+}
+
+# Version of the standard to test against
+# Current options: v1.0, v1.1, v1.2
+VERSION=${VERSION:-"v1.2"}
+
+# Which commit of the standard's repo to use
+# Defaults to the last commit of the 1.2.1_proposed branch
+COMMIT=${COMMIT:-"1.2.1_proposed"}
+
+# Comma-separated list of test names that should be excluded from execution
+# Defaults to "docker_entrypoint, inplace_update_on_file_content"
+EXCLUDE=${EXCLUDE:-"docker_entrypoint,modify_file_content"}
+
+# Additional arguments for the pytest command
+# Defaults to none
+# PYTEST_EXTRA=
+
+# The directory where this script resides
+SCRIPT_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Download archive from GitHub
+if [[ "${VERSION}" = "v1.0" ]] ; then
+  REPO="common-workflow-language"
+else
+  REPO="cwl-$VERSION"
+fi
+
+if [ ! -d "${REPO}-${COMMIT}" ] ; then
+  if [ ! -f "${COMMIT}.tar.gz" ] ; then
+	  wget "https://github.com/common-workflow-language/${REPO}/archive/${COMMIT}.tar.gz"
+  fi
+  tar xzf "${COMMIT}.tar.gz"
+fi
+
+# Setup environment
+venv cwl-conformance-venv
+pip install -U setuptools wheel pip
+pip install -r "${SCRIPT_DIRECTORY}/requirements.txt"
+pip install -r "${SCRIPT_DIRECTORY}/test-requirements.txt"
+
+# Set conformance test filename
+if [[ "${VERSION}" = "v1.0" ]] ; then
+  CONFORMANCE_TEST="${SCRIPT_DIRECTORY}/${REPO}-${COMMIT}/${VERSION}/conformance_test_v1.0.yaml"
+else
+  CONFORMANCE_TEST="${SCRIPT_DIRECTORY}/${REPO}-${COMMIT}/conformance_tests.yaml"
+fi
+
+# Build command
+TEST_COMMAND="python -m pytest ${CONFORMANCE_TEST} -n auto -rs"
+if [[ -n "${EXCLUDE}" ]] ; then
+  TEST_COMMAND="${TEST_COMMAND} --cwl-exclude ${EXCLUDE}"
+fi
+TEST_COMMAND="${TEST_COMMAND} --cov --cov-config ${SCRIPT_DIRECTORY}/.coveragerc --cov-report= ${PYTEST_EXTRA}"
+
+# Create conftest.py
+cat > "$(dirname "${CONFORMANCE_TEST}")/conftest.py" <<EOF
```

Comment:
Why dynamic? Wouldn't it be better to have this as a separate file that you copy in?

Reply:
I wanted to avoid generate confusions to both users and pytest itself with a spurious `conftest.py` file.
For now it's content is pretty simple so it can be autogenerated. If in the future it will become more complicated, probably will be better to move it

Reply:
you could stick it in a hidden or obscure directory :-)
You might enjoy type-checking, formatting, etc...

Reply:
Did it. But not for type checking XD